starknet/src/server/stream.rs

```diff
@@ -74,6 +79,31 @@ where
 
         ResponseStream::new(data_stream).instrument(stream_span)
     }
+
+    async fn sync_status(self, rpc_url: Url) -> StatusResponse {
+        let provider = HttpProvider::new(rpc_url);
+        let latest_apibara_block = self.storage.highest_accepted_block().unwrap().unwrap();
+        let latest_starknet_block = provider.get_head().await.unwrap();
```

Comment:
```suggestion
        let latest_starknet_block = provider.get_head().await.map_err(|_| tonic::Status::internal("internal server error"))?;
```

Don't unwrap the error, but create a new internal server error and return that instead.